include/seqan3/core/configuration/configuration.hpp

```diff
@@ -227,167 +245,6 @@ class configuration : public std::tuple<configs_t...>
     }
     //!\}
 
-    /*!\name Pipe operator
-     * \{
-     */
-    /*!\brief Combines two seqan3::pipeable_config_element objects to a seqan3::configuration.
-     * \tparam lhs_derived_t The derived type of the left hand side operand.
-     * \tparam lhs_value_t   The value type of the left hand side operand.
-     * \tparam rhs_derived_t The derived type of the right hand side operand.
-     * \tparam rhs_value_t   The value type of the right hand side operand.
-     * \param[in] lhs        The left hand operand.
-     * \param[in] rhs        The right hand operand.
-     * \returns A new seqan3::configuration containing `lhs` and `rhs`.
-     */
-    template <typename lhs_derived_t, typename lhs_value_t, typename rhs_derived_t, typename rhs_value_t>
-    friend constexpr auto operator|(pipeable_config_element<lhs_derived_t, lhs_value_t> && lhs,
```

Comment:
I noticed that all of these friend declarations have been removed. Could you please explain why the new implementation of combining configuration objects doesn't use friend declarations?

Reply:
When this class and the pipe-operator was initially implemented, we used a base class to distinguish a configuration object from any other object. This was done via a base class, which allows us to define the operator interface via a reference to the base class and only classes derived from this base class can be used. However, we needed the actual type of the derived class and thus used a CRTP design patterh to get from the base class back to the derived class. Accordingly, we needed all combinations of the |-operator, which were quite a lot.
Instead, we now just use a concept that checks if a type is a base of `configuration_element_base` similar to how views are enabled by inheriting from a view_base class. These classes are empty and do nothing except providing a strong type that allows to distinguish certain types from other objects. Using a concept, however, we could reduce all but one definition of a |-operator. This operator is now a free-function because it uses the public append interface of the configuration class and does not need access to the private internals of the configuration class. That's why it is not a friend function anymore. This makes sense, since the configuration class can also be modified via member functions (which came in later btw). The pipe-operator is simply a neat trick to make the user interface a little more user-friendly.
So both things are now possible:

```cpp
configuration config1 = configuration{align_cfg::method_global{}}.append(align_cfg::band_fixed_size{...}).append(align_cfg::output_score{});
configuration config2 = align_cfg::method_global{} | align_cfg::band_fixed_size{...} | align_cfg::output_score{};
```

Reply:
Thank you! I appreciate the explanation

Reply:
You are welcome.